src/netaddress.cpp

```diff
@@ -272,13 +275,16 @@ bool CNetAddr::IsValid() const
     // header20 vectorlen3 addr26 addr26 addr26 header20 vectorlen3 addr26 addr26 addr26...
     // so if the first length field is garbled, it reads the second batch
     // of addr misaligned by 3 bytes.
-    if (IsIPv6() && memcmp(ip, pchIPv4+3, sizeof(pchIPv4)-3) == 0)
+    if (IsIPv6() && memcmp(m_addr.data(), IPV4_IN_IPV6_PREFIX.data() + 3,
+                           sizeof(IPV4_IN_IPV6_PREFIX) - 3) == 0) {
         return false;
+    }
 
     // unspecified IPv6 address (::/128)
     unsigned char ipNone6[16] = {};
-    if (IsIPv6() && memcmp(ip, ipNone6, 16) == 0)
+    if (IsIPv6() && memcmp(m_addr.data(), ipNone6, sizeof(ipNone6)) == 0) {
```

Comment:
This could be
```C++
    static prevector<16, uint8_t> ipNone6{16, 0x0};
    if (IsIPv6() && m_addr == ipNone6) {
```

Reply:
Right! I will apply it if I modify this PR.

`ADDR_IPV6_SIZE` is of type `size_t`, the typecast should not be needed?

Reply:
Yeah, I had originally put 16 there but realized `ADDR_IPV6_SIZE` was better.